com.unity.render-pipelines.universal/ShaderLibrary/Particles.hlsl

```diff
@@ -110,4 +111,72 @@ half3 SampleNormalTS(float2 uv, float3 blendUv, TEXTURE2D_PARAM(bumpMap, sampler
 #endif
 }
 
+half4 GetParticleColor(half4 color)
+{
+#if defined(UNITY_PARTICLE_INSTANCING_ENABLED)
+#if !defined(UNITY_PARTICLE_INSTANCE_DATA_NO_COLOR)
+    UNITY_PARTICLE_INSTANCE_DATA data = unity_ParticleInstanceData[unity_InstanceID];
+    color = lerp(half4(1.0, 1.0, 1.0, 1.0), color, unity_ParticleUseMeshColors);
+    color *= float4(data.color & 255, (data.color >> 8) & 255, (data.color >> 16) & 255, (data.color >> 24) & 255) * (1.0 / 255);
```

Comment:
I suppose this is the same case as the matrix invert. I think it would be good to add a comment/TODO to replace with library implementation when available.

Reply:
ah, i missed this comment when i read over the PR feedback, sorry. i think this one is actually more clear-cut than the matrix one - i don't see a reason why i cant move it to the Packing file. I'll sort that out :)